central/postgres/pruning.go

```diff
@@ -85,3 +110,36 @@ func GetOrphanedPodIDs(ctx context.Context, pool postgres.DB) ([]string, error)
 		return getOrphanedIDs(ctx, pool, getAllOrphanedPods)
 	})
 }
+
+// GetOrphanedProcessIndicatorIDs returns the pod IDs for process indicators that are orphaned so they can be removed
+func GetOrphanedProcessIndicatorIDs(ctx context.Context, pool postgres.DB, orphanWindow time.Duration) ([]string, error) {
+	return pgutils.Retry2(func() ([]string, error) {
+		ctx, cancel := context.WithTimeout(ctx, orphanedTimeout)
+		defer cancel()
+
+		query := fmt.Sprintf(getAllOrphanedProcesses, int(orphanWindow.Minutes()), int(orphanWindow.Minutes()))
+		return getOrphanedIDs(ctx, pool, query)
+	})
+}
+
+// PruneOrphanedProcessIndicators prunes orphaned process indicators and process listening on ports
+func PruneOrphanedProcessIndicators(ctx context.Context, pool postgres.DB, orphanWindow time.Duration) {
+	// Delete processes listening on ports orphaned because process indicators are orphaned due to
+	// missing deployments
+	query := fmt.Sprintf(deleteOrphanedPLOPDeployments, int(orphanWindow.Minutes()))
```

Comment:
Should we wrap all this in a single transaction?  I did not because I figured some cleanup was better than no cleanup.

Reply:
This makes sense to me, it doesn't create data inconsistencies